### ISSUE TITLE
parameter: P cannot be JSON unwrapped. OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class 

### ISSUE THREAD

**Issue by cforce** [stoicflame/enunciate#208]
This error Message is kind of unlcear and oesn't help to find the cause. The code worked well together with enunciate 1.x, so it mus a configuration maven setup issue

Btw. the java.lang.NoClassDefFoundError: com/webcohesion/enunciate/modules/jaxb/model/util/JAXBUtil
comes although / even i add to my pom (which shall not be necesarry so my understanding from examples for cxf,jaxrs, jackson2 stack)

         <dependency>
            <groupId>com.webcohesion.enunciate</groupId> 
             <artifactId>enunciate-jaxb</artifactId>
         </dependency>

```java
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.1.1:docs (docs) on project vhp-enduser-api: Execution docs of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.1.1:docs failed: com.webcohesion.enunciate.EnunciateException: parameter: P cannot be JSON unwrapped. OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :vhp-enduser-api
Exception in thread "pool-1-thread-1" java.lang.NoClassDefFoundError: com/webcohesion/enunciate/modules/jaxb/model/util/JAXBUtil
	at com.webcohesion.enunciate.modules.jaxb.model.util.MapType.findMapTypeDeclaration(MapType.java:133)
	at com.webcohesion.enunciate.modules.jaxb.model.util.MapType.findMapType(MapType.java:86)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:824)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext$ReferencedTypeDefinitionVisitor.visitDeclared(EnunciateJaxbContext.java:805)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:703)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.addReferencedTypeDefinitions(EnunciateJaxbContext.java:465)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:445)
	at com.webcohesion.enunciate.modules.jaxb.EnunciateJaxbContext.add(EnunciateJaxbContext.java:424)
	at com.webcohesion.enunciate.modules.jaxb.JaxbModule.addPotentialJaxbElement(JaxbModule.java:110)
	at com.webcohesion.enunciate.modules.jaxws.JaxwsModule.call(JaxwsModule.java:152)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: com.webcohesion.enunciate.modules.jaxb.model.util.JAXBUtil
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	... 29 more
```

**Comment by stoicflame** [comment]
Okay, thanks for the report. Assigning to Enunciate 2.2.

**Comment by cforce** [comment]
if i add 
<dependency>
            <groupId>com.webcohesion.enunciate</groupId>
             <artifactId>enunciate-jaxb</artifactId>
</dependency>
 i still get

[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.1.1:docs (docs) on project api: Execution docs of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.1.1:docs failed: com.webcohesion.enunciate.EnunciateException: parameter: P cannot be JSON unwrapped. OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class -> [Help 1]

**Comment by stoicflame** [comment]
Fixed at 9f36be7.

**Comment by stoicflame** [comment]
[Enunciate 2.2 is now available](https://github.com/stoicflame/enunciate/releases/tag/v2.2.0).
